testdata/cheats/Cheats.sol

```diff
@@ -226,4 +226,24 @@ interface Cheats {
     function rpcUrls() external returns (string[2][] memory);
     function parseJson(string calldata, string calldata) external returns (bytes memory);
     function parseJson(string calldata) external returns (bytes memory);
+
+    function serializeBool(string calldata, string calldata, bool) external returns(string memory);
```

Comment:
okay, I've understood how this works now.

providing an identifier key is not really ideal imo.

In order to serialize nested objects, you need to provide the path yourself, right?

This also limits serializing/deserializing to json only, right?

Having a way to abstract this away would be ideal.

that's what I tried with the `Serializer` approach, which would also make the identifier value unnecessary, but is more effort to implement.

I'm hesitant to merge this as is and then make breaking changes later.

I really would like a special json Serializer contract instead.

Internally all the serialize functions will work similar, but this way we can get rid of this identifier, if for example a `jsonSerializer() returns Serializer` cheat code returns a contract which case a new internal id, so instead of `vm.serializeAddress(id, key, value)` you can just call `serializer.serializeAddress(key, value)`


But perhaps I'm overthinking this and this is good enough, wdyt @gakonst?

Reply:
I agree100%. The design assumes this contract to be included in forge-std:

```
contract Serializer {

    string jsonKey;

    address constant HEVM_ADDRESS = address(bytes20(uint160(uint256(keccak256("hevm cheat code")))));

    Cheats vm = Cheats(HEVM_ADDRESS);

    constructor(string memory key){
        jsonKey = key;
    }

    function serialize(string calldata key, bool value) external returns(string memory){
        return vm.serializeBool(jsonKey, key, value);
    }

    function serialize(string calldata key, bool[] memory value) external returns(string memory){
        return vm.serializeBool(jsonKey, key, value);
    }


    function serialize(string calldata key, uint256 value)external returns(string memory){
        return vm.serializeUint(jsonKey, key, value);
    }

    function serialize(string calldata key, uint256[] memory value)external returns(string memory){
        return vm.serializeUint(jsonKey, key, value);
    }


    function serialize(string calldata key, int256 value)external returns(string memory){
        return vm.serializeInt(jsonKey, key, value);
    }

    function serialize(string calldata key, int256[] memory value)external returns(string memory){
        return vm.serializeInt(jsonKey, key, value);
    }

    function serialize(string calldata key, address value)external returns(string memory){
        return vm.serializeAddress(jsonKey, key, value);
    }

    function serialize(string calldata key, address[] memory value)external returns(string memory){
        return vm.serializeAddress(jsonKey, key, value);
    }

    function serialize(string calldata key, bytes32 value)external returns(string memory){
        return vm.serializeBytes32(jsonKey, key, value);
    }

    function serialize(string calldata key, bytes32[] memory value)external returns(string memory){
        return vm.serializeBytes32(jsonKey, key, value);
    }

    function serialize(string calldata key, string memory value)external returns(string memory){
        return vm.serializeString(jsonKey, key, value);
    }

    function serialize(string calldata key, string[] memory value)external returns(string memory){
        return vm.serializeString(jsonKey, key, value);
    }

    function write(string calldata path) external {
        vm.writeJson(jsonKey, path);
    }
}
```